### PR TITLE
fido2luks: 0.2.19 -> 0.2.20, allow discards

### PR DESCRIPTION
**Pull request by danc86**
###### Description of changes

Fido2luks 0.2.20 adds a new option `--allow-discards`. This PR upgrades fido2luks, and also updates the Nixos module to pass `--allow-discards` when the corresponding `allowDiscards` setting is true.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by prusnak** [comment]
Please change commit message from `allow discards with fido2luks` to `fido2luks: pass allow discards option`

**Comment by danc86** [comment]
The second commit is touching the `luksroot` Nixos module, not the `fido2luks` package, so maybe I should prefix the commit message with `nixos/luksroot: `?

**Comment by prusnak** [comment]
> The second commit is touching the `luksroot` Nixos module, not the `fido2luks` package, so maybe I should prefix the commit message with `nixos/luksroot: `?

Yes, please.

**Comment by danc86** [comment]
I amended the second commit to add a `nixos/luksroot:` prefix to the commit message.

**Review by prusnak — APPROVED** [review]
ACK on `pkgs/tools/security/fido2luks/default.nix` part

`nixos/modules/system/boot/luksroot.nix` looks good as well, but I have no way to test.

File: pkgs/tools/security/fido2luks/default.nix [hunk 1]
```diff
@@ -9,13 +9,13 @@
...
   pname = "fido2luks";
-  version = "0.2.19";
+  version = "0.2.20";
 
   src = fetchFromGitHub {
     owner = "shimunn";
     repo = pname;
-    rev = version;
-    sha256 = "sha256-o21KdsAE9KznobdMMKfVmVnENsLW3cMZjssnrsoN+KY=";
+    rev = "210da1ce0fdd8a7259329b9cff636b4ab9929936";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/fido2luks/default.nix:17]: https://github.com/shimunn/fido2luks/tree/0.2.20

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
.

File: pkgs/tools/security/fido2luks/default.nix (hunk 1 above)
Comment (prusnak) [reply pkgs/tools/security/fido2luks/default.nix:17]: Please keep `rev = version;` - the referenced commit has been tagged as `0.2.20` so it should work correctly.

Comment (danc86) [reply pkgs/tools/security/fido2luks/default.nix]: Hmm yeah, not sure what I was thinking there. I've amended the first commit to keep `rev = version;`.

**Review by prusnak — APPROVED** [review]

**Pull request merged**